[Term Entry]  Javascript Method :  .toLocaleString() (#4608)
Codecademy/docs: content/javascript/concepts/dates/terms/toLocaleString/toLocaleString.md

---

> Title: '.toLocaleString()'
> Description: 'Returns a string formatted based on the locale for a Date object in JavaScript.'
> Subjects:
>   - 'Computer Science'
>   - 'Web Development'
> Tags:
>   - 'Date'
>   - 'Methods'
>   - 'Strings'
> CatalogContent:
>    - 'introduction-to-javascript'
>    - 'paths/front-end-engineer-career-path'

Reviewer: ```suggestion
Title: '.toLocaleString()'
Description: 'Returns a string formatted based on the locale for a Date object in JavaScript.'
Subjects:
  - 'Computer Science'
  - 'Web Development'
Tags:
  - 'Date'
  - 'Methods'
  - 'Strings'
CatalogContent:
  - 'introduction-to-javascript'
  - 'paths/front-end-engineer-career-path'
```

---

> In JavaScript, the `.toLocaleString()` method formats a [Date](https://www.codecademy.com/resources/docs/javascript/dates) object as a string according to the specified locale, considering cultural settings such as `language` and `date/time` formatting preferences specific to the chosen region or country.

Reviewer: ```suggestion
In JavaScript, the `.toLocaleString()` method formats a [`Date`](https://www.codecademy.com/resources/docs/javascript/dates) object as a string according to the specified locale, considering cultural settings such as `language` and `date/time` formatting preferences specific to the chosen region or country.
```

## Syntax  

> ```js

Reviewer: ```suggestion
```pseudo
```
The syntax should be inside a pseudo block.

dateObj.toLocaleString(locales, options)
```

- `dateObj`: A `Date` object representing the date and time to be formatted as a string based on the specified locale.
- `locales`: A `string` or an `array of strings` that specifies one or more `locales` or language tags for formatting the date. 
- `options`: An object that allows customizing the formatting behavior, such as specifying the format for `date`, `time`, `numeric values`, and more.

> **Note:** Both the `locales` and `options` parameters in the `.toLocaleString()` method are optional. If the parameters are not provided, the method will use `default` values based on the runtime environment.

> 

Reviewer: ```suggestion
```

## Example  

> In the example below, `toLocaleString()` formats the current `date` and `time` according to the long date format with the full `weekday`, `month`, `day`, and `year` in English (United States) `locale`.

Reviewer: ```suggestion
In the example below, `toLocaleString()` formats the current `date` and `time` according to the long date format with the full `weekday`, `month`, `day`, and `year` in English (United States) `locale`:
```

> ```javascript {copy}

Reviewer: ```suggestion
```js
```

> const currentDate = new Date(); 
> const locale = 'en-US';
> const options = { weekday: 'long', year: 'numeric', month: 'long', day: 'numeric' };
> console.log(currentDate.toLocaleDateString(locale, options));

Reviewer: ```suggestion
const specificDate = new Date('2024-05-16'); 
const locale = 'en-US';
const options = { weekday: 'long', year: 'numeric', month: 'long', day: 'numeric' };
console.log(specificDate.toLocaleDateString(locale, options));
```
The problem with this code is that the output would vary depending on the current date and the viewer will might get confused. Let's just specify the date so the output will be the same every time.

```

The code above produces the following output:

```shell

> Monday, May 6, 2024

Reviewer: ```suggestion
Thursday, May 16, 2024
```

```

## Codebyte Example

> In the following example, `.toLocaleString()` formats the current date and time with the full `weekday`, `year`, `month`, `day`, `hour`, `minute`, and `second` in French (Morocco) locale (fr-MA).

Reviewer: ```suggestion
In the following example, `.toLocaleString()` formats the current date and time with the full `weekday`, `year`, `month`, `day`, `hour`, `minute`, and `second` in French (Morocco) locale (fr-MA):
```

---
Title: 'toLocaleString Method'
Description: 'A method to format a Date object as a string, based on the locale.'
Subjects:
  - 'Computer Science'
  - 'Web Development'
  
Tags:
  - 'JavaScript'
  - 'Methods'
  - 'Date'
  
CatalogContent:
  - 'learn-javascript'
  - 'paths/web-development'
---

# toLocaleString() Method :

The `.toLocaleString()` method in JavaScript is used to format a [`Date` object](https://www.codecademy.com/resources/docs/javascript/dates), as a string based on the specified locale. This means it takes into account cultural settings such as `language` and `date/time` formatting preferences specific to the selected region or country.

> ## Syntax : 

Reviewer 2: ```suggestion
## Syntax
```

```js {copy}
toLocaleString(locales [, options])
```

> - `locales` : Represents the `locale or locales` to use when formatting the `date`.
> - `options` : Contains additional formatting options for the `date`, such as the format of the `weekday`, `month`, `year`, etc.
> 

Reviewer 2: ```suggestion

- `dateObj`: The `Date` object used to format as a string based on the locale.
- `locales`: A `string` or an `array of strings` that specifies one or more `locales` or language tags for formatting the date. 
- `options`: An object that allows customizing the formatting behavior, such as specifying the format for `date`, `time`, `numeric values`, and more.

> **Note:** Both the `locales` and `options` parameters in the `.toLocaleString()` method are optional. If the parameters are not provided, the method will use `default` values based on the runtime environment.
 ```

> ## Example : 

Reviewer 2: ```suggestion
## Example
```

```js {copy}
const date = new Date(); // Current date and time
const options = { weekday: 'long', year: 'numeric', month: 'long', day: 'numeric' };
console.log(date.toLocaleString('en-US', options));

> ```

Reviewer 2: Please modify this example, the code has been plagiarized.
Plagiarised from: `https://debbie.codes/blog/date-formatting/`

In this example, `toLocaleString()` formats the current `date` and `time` according to the long date format with the full `weekday`, `month`, `day`, and `year` in English (United States) `locale`.
The output could be something like: 
```shell {copy}
// Output : "Sunday, May 5, 2024"

> ```

Reviewer 2: Please modify the output and its description based on the new code example that you will use.

> Depending on the current `date` and `time` when the code is executed.
> 

Reviewer 2: ```suggestion
```

## Runnable Code Example : 
```js {copy}
const date = new Date(1985,7,4);  
const options = { weekday: 'long', year: 'numeric', month: 'long', day: 'numeric' };

// Formatting for English (United States) :
console.log(date.toLocaleString('en-US', options));

// Formatting for Korean (South Korea) :
console.log(date.toLocaleString('ko-KR', options));

// Formatting for Hindi (India) : 
console.log(date.toLocaleString('hi-IN', options));
```
In this example, we use `toLocaleString()` to format the current `date` and `time` with the specified formatting options for `English` (United States), `Korean` (South Korea), and `Hindi` (India).

```shell {copy}
// Output : 
For English (United States): "Sunday, August 4, 1985" .
For Korean (South Korea):  "1985년 8월 4일 일요일" .
For Hindi (India): "रविवार, 4 अगस्त 1985" .
```

> 

Reviewer 2: The lines 45 to 70 can be removed. Only one ## Example and one ## Codebyte Example is needed.

## Codebyte Example

In the following example the variables `locale` and `options` can be modified, to print the `Date` in a custom format:

> ```codebyte/js

Reviewer 2: ```suggestion
```codebyte/javascript
```

const date = new Date(Date.now());

const locale = 'fr-MA';
const options = {
  year: 'numeric',
  month: 'long',
  day: 'numeric',
  weekday: 'long',
};

> console.log(date.toLocaleDateString(locale, options));

Reviewer 2: ```suggestion
console.log(date.toLocaleString(locale, options));
```

```

> 
> 

Reviewer 2: ```suggestion
```
